### PR TITLE
RELATED: ONE-4642 List components in ui-kit

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
New List components in ui-kit, supporting custom primary color. 
All components used from GS were moved to ui-kit + divided to the Legacy ones and new ones, which should be used in future

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by ivan-nejezchleb** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1171/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/403/ : FAILURE in 4m 05s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1170/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2775/ : FAILURE in 4m 28s

**Comment by ivan-nejezchleb** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1172/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2776/ : SUCCESS in 19m 56s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1173/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/404/ : SUCCESS in 25m 28s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1182/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2785/ : FAILURE in 5m 24s

**Comment by ivan-nejezchleb** [comment]
extended test

**Comment by ivan-nejezchleb** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1188/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2789/ : SUCCESS in 9m 45s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1192/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2794/ : FAILURE in 2m 36s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1193/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2795/ : SUCCESS in 12m 07s

File: examples/sdk-examples/package.json [hunk 1]
```diff
@@ -82,7 +82,7 @@
     },
     "dependencies": {
         "@gooddata/api-client-bear": "^8.1.0-alpha.11",
-        "@gooddata/goodstrap": "^68.20.4",
+        "@gooddata/goodstrap": "69.0.0",
```
Comment (no23reason) [inline examples/sdk-examples/package.json:85]: Please use carets

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-kit/src/List/InsightListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                    </div>
+                </div>
+                {this.renderActions()}
+            </div>
+        );
+    }
+
+    public componentDidUpdate(prevProps: IInsightListItemProps): void {
+        if (prevProps.width !== this.props.width && this.shortenedTextRef.current) {
+            // TODO INE remove any when GD has correct typings
```
Comment (no23reason) [inline libs/sdk-ui-kit/src/List/InsightListItem.tsx:95]: Is this actionable now after the Goodstrap 69 release?

File: libs/sdk-ui-kit/src/List/LegacySingleSelectList.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+ * @internal
+ * @deprecated This component is deprecated use SingleSelectList instead
+ */
+export class LegacySingleSelectList<T> extends Component<ILegacySingleSelectListProps<T>> {
+    static defaultProps = {
+        className: "",
+        getItemKey: guidFor,
+        items: [] as any[],
+        itemsCount: 0,
+        listItemClass: LegacySingleSelectListItem, // TODO add tests
```
Comment (no23reason) [inline libs/sdk-ui-kit/src/List/LegacySingleSelectList.tsx:45]: Is this TODO actionable? Or are the tests necessary now that it is deprecated?

File: examples/sdk-examples/package.json (hunk 1 above)
Comment (ivan-nejezchleb) [reply examples/sdk-examples/package.json:85]: fixed

File: libs/sdk-ui-kit/src/List/InsightListItem.tsx (hunk 2 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-kit/src/List/InsightListItem.tsx:95]: Unluckily this issue was not fixed in new GS :(

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-kit/src/List/LegacySingleSelectList.tsx (hunk 3 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-kit/src/List/LegacySingleSelectList.tsx]: removed as you mentioned it is now Legacy component

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline gate aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/442/ : FAILURE in 7m 25s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/700/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2806/ : SUCCESS in 18m 20s

**Pull request merged**